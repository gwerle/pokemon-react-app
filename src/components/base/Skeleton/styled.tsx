import styled from "styled-components";

type SkeletonContainerProps = {
  height: string;
  width: string;
};

export const SkeletonContainer = styled.div<SkeletonContainerProps>`
  background: #eee;
  background: linear-gradient(110deg, #ececec 8%, #f5f5f5 18%, #ececec 33%);
  border-radius: 5px;
  background-size: 200% 100%;
  height: ${props => props.height};
  width: ${props => props.width};
`;
