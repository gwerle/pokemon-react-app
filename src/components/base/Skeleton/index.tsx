import * as S from "./styled";

type Props = {
  width: string;
  height: string;
};

export default function Skeleton({ height, width }: Props) {
  return (
    <S.SkeletonContainer width={width} height={height} data-testid="skeleton" />
  );
}
