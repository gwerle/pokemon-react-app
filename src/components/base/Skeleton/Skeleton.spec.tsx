import { renderWithTheme } from "@/utils/tests/renderWithTheme";

import Skeleton from ".";

describe("<Skeleton />", () => {
  it("Renders without error", () => {
    const { container } = renderWithTheme(
      <Skeleton width="100px" height="100px" />
    );

    expect(container.firstChild).toHaveStyle("height: 100px");
    expect(container.firstChild).toHaveStyle("width: 100px");
    expect(container).toMatchSnapshot();
  });
});
