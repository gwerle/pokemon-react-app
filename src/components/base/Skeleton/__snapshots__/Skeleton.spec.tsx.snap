// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> Renders without error 1`] = `
<div>
  <div
    class="sc-beqWaB iuOWuC"
    data-testid="skeleton"
    height="100px"
    width="100px"
  />
</div>
`;
